src/dmd/parse.d

```diff
@@ -2152,6 +2154,35 @@ final class Parser(AST) : Lexer
                             cppmangle = token.value == TOK.class_ ? CPPMANGLE.asClass : CPPMANGLE.asStruct;
                             nextToken();
                         }
+                        else if (token.value == TOK.string_) // extern(C++, "namespaces")
+                        {
+                            ignoreCppSymbols = true;
+                            idents = new AST.Identifiers();
+                            AST.StringExp stringExp = cast(AST.StringExp)parsePrimaryExp();
+                            char[] names = stringExp.toStringz();
+                            uint startIndex = 0;
+                            while(startIndex < names.length)
+                            {
+                                uint i = startIndex;
+                                foreach(ref char c ; names[startIndex .. $])
+                                {
+                                    i += 1;
+                                    if (c == '.')
+                                    {
+                                        if (i != names.length)
+                                        {
+                                            c = '\0';
+                                            break;
+                                        }
+                                        else
+                                            error("C++ namespace expected after `.` at end of `%s`", names.ptr);
+
+                                    }
+                                }
+                                idents.push(new Identifier(&names[startIndex]));
```

Comment:
Send a slice to `new Identifier` instead, then you don't need to make things mutable.

Reply:
I was going to, but see my other comment.

Reply:
NVM, seems idPool() is the correct way to create a new Identifer.